File: Unit4L.go
// Author: William Provost
// Version: 1.0.0
// Date: 2025-12-07
// Fileoverview: This program will simulate the rolling of a 10-sided die.

package main

import (
	"fmt"
	"math/rand/v2" // import the random package, new in Go 1.22+
)

func main() {
	// constants
	const MIN int = 1
	const MAX int = 10
	
	// This removes the S1021 warning.
	roll := rand.IntN(MAX-MIN+1) + MIN 

	// display results
	fmt.Println("Rolling a 10-sided die...")
	fmt.Printf("You rolled a %d\n", roll)

	fmt.Println("\nDone.")
}

File: Unit4L.ts
/**
 * @author William Provost
 * @version 1.0.0
 * @date 2025-12-07
 * @fileoverview This program will simulate the rolling of a 10-sided die.
 */

// constants and variables
const MIN = 1;
const MAX = 10;
let roll: number;

// roll the die
// Math.random() generates a float in [0, 1).
// (MAX - MIN + 1) is 10. Multiplying gives [0, 10).
// Math.floor() gives integer [0, 9]. Adding MIN (1) shifts range to [1, 10].
roll = Math.floor(Math.random() * (MAX - MIN + 1) + MIN);

// display results
console.log("Rolling a 10-sided die...");
console.log(`You rolled a ${roll}.`);

console.log("\nDone.");
